addons/website_slides/models/res_users.py

```diff
@@ -13,17 +13,17 @@ def create(self, vals_list):
         users = super(Users, self).create(vals_list)
         for user in users:
             self.env['slide.channel'].sudo().search([
-                ('enroll_group_ids', 'in', user.groups_id.ids)
+                ('enroll_group_ids', 'in', user.group_ids.all_implied_ids.ids)
             ])._action_add_members(user.partner_id)
         return users
 
     def write(self, vals):
         """ Trigger automatic subscription based on updated user groups """
         res = super(Users, self).write(vals)
-        sanitized_vals = self._remove_reified_groups(vals)
-        if sanitized_vals.get('groups_id'):
-            added_group_ids = [command[1] for command in sanitized_vals['groups_id'] if command[0] == 4]
-            added_group_ids += [id for command in sanitized_vals['groups_id'] if command[0] == 6 for id in command[2]]
+        if 'group_ids' in vals:
+            group_ids = [command[1] for command in vals['group_ids'] if command[0] == 4]
+            group_ids += [id for command in vals['group_ids'] if command[0] == 6 for id in command[2]]
+            added_group_ids = self.env['res.groups'].browse(group_ids).all_implied_ids.ids
```

Comment:
Avoid using magic numbers and `id` would be masking the python's default function.
```suggestion
            group_ids = [command[1] for command in vals['group_ids'] if command[0] == Command.LINK]
            group_ids += [id_ for command in vals['group_ids'] if command[0] == Command.SET for id_ in command[2]]
            added_group_ids = self.env['res.groups'].browse(group_ids).all_implied_ids.ids
```